CorsixTH/Lua/queue.lua

```diff
@@ -313,16 +313,22 @@ function Queue:movePatient(index, new_index)
 end
 
 --! Called when reception desk is destroyed, or when a room is destroyed from a crashed machine.
-function Queue:rerouteAllPatients(action)
+--!param room_id Id of the room to reroute to, 'nil' for reception.
+function Queue:rerouteAllPatients(room_id)
   for _, humanoid in ipairs(self) do
     -- check by class type as staff/vips shouldn't get a SeekRoomAction
     if class.is(humanoid, Patient) then
       -- slight delay so the desk is really destroyed before rerouting
```

Comment:
I just realised this comment is slightly poor in context. I can get what it means at least, so maybe nothing to do/scope.

Reply:
Out of scope here, but imho the solution is poor. Instead of delaying and hoping it works, find a replacement action to perform. (Eg by finding another available desk or lacking that, start idling.)

This sort of hack is also done with the receptionist when you grab the desk.

Basically it means an NPC cannot start looking for a new desk by itself. Instead a desk to aim for needs to be provided externally.

Reply:
#1681 was closed a while back, but that was my attempt at trying to fix some reception bugs. I could probably make it cleaner now, though.